### PR TITLE
Check that vm has no snapshots before changing service offering

### LINKED ISSUE #1492: Inform a user that VM snapshots will be discarded if he/she changes service offering
This issue covers the following requirements:
1. [REQ_VM_DTLS_055] A system should inform user that VM snapshots will be discarded if he changes service offering

### PR DESCRIPTION
**Pull request by andrewbents**
closes #1492 

snapshots

### CODE REVIEW

**Comment by bwsw** [comment]
@wowshakhov 
Implemented as agreed. Please review.

File: src/i18n/ru.json [hunk 1]
```diff
@@ -75,7 +75,8 @@
       "CONFIRM_STOP": "Вы уверены, что хотите остановить виртуальную машину?",
       "CONFIRM_REBOOT": "Вы уверены, что хотите перезапустить виртуальную машину?",
       "CONFIRM_RESTORE": "Вы уверены, что хотите переустановить виртуальную машину?",
-      "REMOVE_SNAPSHOTS": "Пожалуйста, удалите снимки ВМ перед переустановкой",
+      "REMOVE_SNAPSHOTS_REINSTALL": "Пожалуйста, удалите снимки ВМ перед переустановкой",
```
Comment (wowshakhov) [inline src/i18n/ru.json]: Do we need a period here after the sentence?

File: src/app/vm/shared/vm-restore-dialog/vm-restore-dialog.component.spec.ts [hunk 2]
```diff
@@ -18,116 +13,42 @@ describe('VmRestoreDialogComponent', () => {
...
-          provide: MAT_DIALOG_DATA,
-          useValue: { vm },
-        },
-        {
-          provide: VmSnapshotService,
-          useValue: snapshotService,
-        },
-      ],
+      declarations: [VmRestoreDialogComponent, MockTranslatePipe, mockMatDialogClose],
+      providers: [],
```
Comment (wowshakhov) [inline src/app/vm/shared/vm-restore-dialog/vm-restore-dialog.component.spec.ts]: Do we need empty providers?

File: src/app/vm/shared/vm-snapshots-checker-dialog/vm-snapshots-checker-dialog.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+  function findContent(): HTMLElement {
+    return fixture.nativeElement.querySelector('.mat-dialog-content');
+  }
+
+  function setReturnedSnapshots(snaps: VmSnapshot[]) {
+    snapshotService.getList.and.returnValue(of(snaps));
+  }
+
+  it('should a loading when snapshots are not fetched', () => {
```
Comment (wowshakhov) [inline src/app/vm/shared/vm-snapshots-checker-dialog/vm-snapshots-checker-dialog.component.spec.ts]: should **a** loading?

File: src/app/vm/shared/vm-snapshots-checker-dialog/vm-snapshots-checker-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import { Component, Inject, OnDestroy, OnInit, Type } from '@angular/core';
+import { MAT_DIALOG_DATA } from '@angular/material';
+import { Subscription } from 'rxjs';
+import { VmSnapshotService } from '../../../shared/services/vm-snapshot.service';
+import { VirtualMachine } from '../vm.model';
+
+export interface VmSnapshotsCheckerDialogData {
+  vm: VirtualMachine;
+  component: Type<any>;
```
Comment (wowshakhov) [inline src/app/vm/shared/vm-snapshots-checker-dialog/vm-snapshots-checker-dialog.component.ts]: Can we use a generic here if we need component's type? If we don't, we can use unknown instead.

File: src/i18n/ru.json (hunk 1 above)
Comment (andrewbents) [reply src/i18n/ru.json]: @bwsw

Comment (bwsw) [reply src/i18n/ru.json]: @andrewbents @wowshakhov that's fine as it is. I'll fix later if necessary.

**Pull request merged**